src/Moryx.AbstractionLayer/Products/ProductIdentity.cs

```diff
@@ -32,6 +32,17 @@ public ProductIdentity(string identifier, short revision)
         /// </summary>
         public static ProductIdentity AsLatestRevision(string identifier) => new ProductIdentity(identifier, LatestRevision);
 
+        /// <summary>
+        /// Create a product identity from a string. Counterpart to ToString() method
+        /// </summary>
+        /// <param name="identifier"></param>
+        /// <returns></returns>
+        public static ProductIdentity Parse(string identifier)
+        {
+            var splitIdentity = identifier.Split('-');
+            return splitIdentity.Length != 2 ? null : new ProductIdentity(splitIdentity[0], Convert.ToInt16(splitIdentity[1]));
```

Comment:
Instead of returning null, I would return a FormatException and provide also a TryParse method like in [int.Parse](https://docs.microsoft.com/de-de/dotnet/api/system.int32.parse?view=net-5.0)

Reply:
Addition: I would name the argument `identityString` or similar. `Identifier` refers only to the first part without the revision in all other occurences.